### PR TITLE
HPCC-20558 Add ability for daliadmin listmatches to return multiple values

### PR DESCRIPTION
**Pull request by jakesmith**
e.g. :
daliadmin <dali-ip> listmatches /WorkUnits 'W201809*' @jobName,State

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by jakesmith** [comment]
@mckellyln - please review

File: dali/daliadmin/daliadmin.cpp [hunk 1]
```diff
@@ -1889,16 +1889,32 @@ static void listmatches(const char *path, const char *match, const char *pval)
     {
         output.append(", match=").append(match);
         if (pval)
-            output.append(", property value = ").append(pval);
+            output.append(", property value(s) = ").append(pval);
     }
+    outln(output);
```
Comment (mckellyln) [inline dali/daliadmin/daliadmin.cpp]: Is this an extra outln() ?

**Comment by mckellyln** [comment]
@jakesmith one question about the outln()

File: dali/daliadmin/daliadmin.cpp (hunk 1 above)
Comment (jakesmith) [reply dali/daliadmin/daliadmin.cpp]: Yes, it was unintentionally missing before, meaning the lines above it were building up a string ready for output, that never was.

File: dali/daliadmin/daliadmin.cpp [hunk 2]
```diff
@@ -109,7 +109,7 @@ void usage(const char *exe)
   printf("\n");
   printf("Workunit commands:\n");
   printf("  listworkunits [<prop>=<val> [<lower> [<upper>]]] -- list workunits that match prop=val in workunit name range lower to upper\n");
-  printf("  listmatches <connection xpath> [<match xpath>=<val> [<property xpath>]]\n");
+  printf("  listmatches <connection xpath> [<match xpath>=<val> [<property xpaths> -- <property xpaths> is comma seperated list of xpaths\n");
```
Comment (jakesmith) [inline dali/daliadmin/daliadmin.cpp]: @mckellyln - just noticed missing ]] here - I'll fix and squash it in.

**Review by mckellyln — APPROVED** [review]
@jakesmith approved

**Comment by mckellyln** [comment]
@jakesmith good to merge

**Comment by jakesmith** [comment]
@mckellyln - fixed missing ]] in usage

**Comment by jakesmith** [comment]
@richardkchapman - please merge

File: dali/daliadmin/daliadmin.cpp [hunk 3]
```diff
@@ -109,7 +109,7 @@ void usage(const char *exe)
   printf("\n");
   printf("Workunit commands:\n");
   printf("  listworkunits [<prop>=<val> [<lower> [<upper>]]] -- list workunits that match prop=val in workunit name range lower to upper\n");
-  printf("  listmatches <connection xpath> [<match xpath>=<val> [<property xpath>]]\n");
+  printf("  listmatches <connection xpath> [<match xpath>=<val> [<property xpaths>]] -- <property xpaths> is comma seperated list of xpaths\n");
```
Comment (richardkchapman) [inline dali/daliadmin/daliadmin.cpp]: Spelling: separated not seperated.

**Comment by richardkchapman** [comment]
@jakesmith One spelling issue

**Pull request merged**